channeldb/graph.go

```diff
@@ -3013,11 +3013,11 @@ func putChanEdgePolicy(edges *bolt.Bucket, edge *ChannelEdgePolicy, from, to []b
 		// In order to delete the old entry, we'll need to obtain the
 		// *prior* update time in order to delete it. To do this, we'll
 		// create an offset to slice in. Starting backwards, we'll
-		// create an offset than puts us right after the flags
-		// variable:
+		// create an offset than puts us right after the flags variable:
 		//
-		//  * pubkeySize + fee+policySize + timelockSize + flagSize
-		updateEnd := 33 + (8 * 3) + 2 + 1
+		//  * pubKeySize + proportionalFeeSize + baseFeeSize +
+		//    minHTLCSize + timeLockDeltaSize + flagSize
+		updateEnd := len(edgeBytes) - (33 + 8 + 8 + 8 + 2 + 2)
```

Comment:
Huh, does this mean this probably would never work? :P

Reply:
Yeah, they were never removed properly due to the incorrect offset calculation.

Reply:
Yeah there's soooo many stray entries in the databases now. We may ever want to instruct select node operators to run the compaction tool afterwards.

Reply:
The test itself also had a flaw that caused this to go unnoticed until now.

Reply:
This should be changed to deserialize the policy, and read the timestamp off the parsed struct. This method of slicing out the timestamp will fail as soon as #1825 lands and we have extra data appended to the policies. I'd rather not have to go through this again :P

Reply:
Fixed!